### PR TITLE
[stable/influxdb] Adds option to create default user to influxDB

### PR DESCRIPTION
**Pull request by jose5918**
Reasons for change:
- When authentication is enabled, at least one admin user must be
  available

Changes made:
- Adds job.yaml template
- This job creates a default user if `setDefaultUser` is enabled in values.yaml

Testing done:
- Deployed in cluster and verified the creation of a user in InfluxDB

### CODE REVIEW

File: stable/influxdb/templates/job.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        app: {{ template "fullname" . }}
+        release: "{{ .Release.Name }}"
+    spec:
+      containers:
+      - name: {{ template "fullname" . }}-create-user
+        image: "{{ .Values.setDefaultUser.image }}"
+        args:
+          - "-X"
+          - POST
+          - "http://{{ template "fullname" . }}:{{ .Values.config.http.bind_address }}/query"
```
Comment (jackzampolin) [inline stable/influxdb/templates/job.yaml]: Does this need to be `https://{{ template "fullname" . }}.{{ .Release.Namespace }}:{{ .Values.config.http.bind_address}}/query`?

File: stable/influxdb/templates/job.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{{- if .Values.setDefaultUser.enabled -}}
```
Comment (jackzampolin) [inline stable/influxdb/templates/job.yaml]: rename file to `post-install-set-auth.yaml`?

File: stable/influxdb/templates/job.yaml (hunk 1 above)
Comment (jackzampolin) [inline stable/influxdb/templates/job.yaml]: Does this need to be `https://{{ template "fullname" . }}.{{ .Release.Namespace }}:{{ .Values.config.http.bind_address }}/query`?

**Comment by jackzampolin** [comment]
@jose5918 This looks great! A couple of small things. Also could you add some info about this the the `stable/influxdb/README.md`?

This has been on my list! Thank you for putting this together.

**Comment by jose5918** [comment]
@jackzampolin I am actually currently working on a change to this. I think it would be considered correct to pull in the values being set from a secret so i will let you know once I have it fixed.

File: stable/influxdb/templates/job.yaml (hunk 1 above)
Comment (jose5918) [reply stable/influxdb/templates/job.yaml]: From my understanding this only needs to be done when across namespaces and the job should remain within the same namespace when deployed by helm as a chart

File: stable/influxdb/templates/job.yaml (hunk 2 above)
Comment (jose5918) [reply stable/influxdb/templates/job.yaml]: Sure

**Comment by jose5918** [comment]
@jackzampolin I have moved the sensitive information to a secret and made your requested changes.

File: stable/influxdb/README.md [hunk 3]
```diff
@@ -70,3 +70,13 @@ $ helm install --name my-release -f values.yaml stable/influxdb
...
+
+## Starting with authentication
+
+In `values.yaml` change `.Values.config.http.auth_enabled` to `true`.
+
+Influxdb requires also a user to be set in order for authentication to be enforced. See more details [here](https://docs.influxdata.com/influxdb/v1.2/query_language/authentication_and_authorization/#set-up-authentication).
+
+To handle this setup on startup, a job can be enabled in `values.yaml` by setting `.Values.setDefaultUser.enabled` to `true`.
+
+Make sure to uncomment or configure the job settings after enabling it.
```
Comment (jackzampolin) [inline stable/influxdb/README.md]: Adding a note that if you don't set a password, a 'random' one will be created would be nice here.

File: stable/influxdb/values.yaml [hunk 4]
```diff
@@ -23,6 +23,40 @@ persistence:
...
+  # restartPolicy: OnFailure
+
+  # user:
+
+    ## The user name
+    ## Default: "admin"
+    # username: "admin"
+
+    ## User password
+    ## Default: "admin"
```
Comment (jackzampolin) [inline stable/influxdb/values.yaml]: This is no longer correct. The password will default to `{{ randAscii 10 }}`. Please be sure to note this in the `values.yaml` file.

**Review by jackzampolin — CHANGES_REQUESTED** [review]

**Comment by jackzampolin** [comment]
A couple more small things. Looks good with the secret!

**Comment by jose5918** [comment]
@jackzampolin Should be good to go

**Comment by jackzampolin** [comment]
@jose5918 LGTM! 👍

**Comment by lachie83** [comment]
@k8s-bot ok to test

**Comment by seanknox** [comment]
Hi @jose5918: can you bump the chart version to `0.3.0`? I'll then merge in. Thanks!

**Comment by jose5918** [comment]
@seanknox Thanks! Should be ready now

**Comment by seanknox** [comment]
Thanks @jose5918!

**Comment by vanto** [comment]
Question: Is there a reason why this has been solved via a post install hook job instead of relying on the environment variable support for this, provided by the [official influxdb docker image](https://hub.docker.com/_/influxdb/)? There are the variables `INFLUXDB_ADMIN_USER` and `INFLUXDB_ADMIN_PASSWORD` defined for the same purpose of this ticket.

**Comment by jose5918** [comment]
@vanto Looks like this was simpler to do at the time. Those environment variable options were not available for the container yet.

They came in a few months later https://github.com/influxdata/influxdata-docker/commit/98b260574b332e654478d64334dfb27513292a42#diff-d4e0a564e2a8ab5a97797191488eada4

Probably a good idea to open an issue and update this

**Pull request merged**